internal/controller/pkg/revision/reconciler.go

```diff
@@ -313,6 +319,56 @@ func SetupConfigurationRevision(mgr ctrl.Manager, o controller.Options) error {
 		Complete(ratelimiter.NewReconciler(name, r, o.GlobalRateLimiter))
 }
 
+// SetupFunctionRevision adds a controller that reconciles FunctionRevisions.
+func SetupFunctionRevision(mgr ctrl.Manager, o controller.Options) error {
+	name := "packages/" + strings.ToLower(v1alpha1.FunctionRevisionGroupKind)
+	nr := func() v1.PackageRevision { return &v1alpha1.FunctionRevision{} }
+
+	clientset, err := kubernetes.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		return errors.Wrap(err, errCannotInitializeHostClientSet)
+	}
+
+	metaScheme, err := xpkg.BuildMetaScheme()
+	if err != nil {
+		return errors.New(errCannotBuildMetaSchema)
+	}
+	objScheme, err := xpkg.BuildObjectScheme()
+	if err != nil {
+		return errors.New(errCannotBuildObjectSchema)
+	}
+	fetcher, err := xpkg.NewK8sFetcher(clientset, append(o.FetcherOptions, xpkg.WithNamespace(o.Namespace), xpkg.WithServiceAccount(o.ServiceAccount))...)
+	if err != nil {
+		return errors.Wrap(err, errCannotBuildFetcher)
+	}
+
+	r := NewReconciler(mgr,
+		WithCache(o.Cache),
+		WithDependencyManager(NewPackageDependencyManager(mgr.GetClient(), dag.NewMapDag, v1beta1.FunctionPackageType)),
+		WithHooks(NewFunctionHooks(resource.ClientApplicator{
+			Client:     mgr.GetClient(),
+			Applicator: resource.NewAPIPatchingApplicator(mgr.GetClient()),
+		}, o.Namespace, o.ServiceAccount)),
+		WithEstablisher(NewAPIEstablisher(mgr.GetClient(), o.Namespace)),
+		WithNewPackageRevisionFn(nr),
+		WithParser(parser.New(metaScheme, objScheme)),
+		WithParserBackend(NewImageBackend(fetcher, WithDefaultRegistry(o.DefaultRegistry))),
+		WithLinter(xpkg.NewFunctionLinter()),
+		WithLogger(o.Logger.WithValues("controller", name)),
+		WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name))),
+	)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.FunctionRevision{}).
+		Owns(&appsv1.Deployment{}).
+		Watches(&v1alpha1.ControllerConfig{}, &EnqueueRequestForReferencingProviderRevisions{
+			client: mgr.GetClient(),
+		}).
```

Comment:
Is `EnqueueRequestForReferencingProviderRevisions` correct here?

Reply:
Nope, it wasn't. I created `EnqueueRequestForReferencingFunctionRevisions` struct and implemented relevant methods.